api/miner.go

```diff
@@ -1,10 +1,30 @@
 package api
 
 import (
+	"encoding/json"
 	"fmt"
 	"net/http"
+
+	"github.com/NebulousLabs/Sia/crypto"
+	"github.com/NebulousLabs/Sia/types"
 )
 
+// MinerBlockforworkResp is the response struct returned by the API call
+// '/miner/blockforwork'.
+type MinerBlockforworkResp struct {
+	Block      types.Block
+	MerkleRoot crypto.Hash
+	Target     types.Target
+}
+
+// minerBlockforworkHandler handles the API call that retrieves a block for
+// work.
+func (srv *Server) minerBlockforworkHandler(w http.ResponseWriter, req *http.Request) {
+	var bfw MinerBlockforworkResp
+	bfw.Block, bfw.MerkleRoot, bfw.Target = srv.miner.BlockForWork()
```

Comment:
why do you need more than just the block? You can calculate the MerkleRoot by calling `.MerkleRoot` on the block, and you can get the target by calling `srv.cs.ChildTarget`. Unless the server and the miner are out of sync, I guess.